### ISSUE TITLE
[Android] Faceid not working on android

### ISSUE THREAD

**Issue by MDG-MMeksasi** [SelfLender/react-native-biometrics#243]
Hello, I'm using Samsung A13 with only faceId enrolled, but it's always returning this error **BIOMETRIC_ERROR_NONE_ENROLLED**  and that biometrics is not available, while if I have a fingerprint enrolled it's working properly.

If anyone got a solution.

    "react-native": "0.69.1",
    "react-native-biometrics": "^3.0.1"
    device: Samsung A13

**Comment by MDG-MMeksasi** [comment]
#241

**Comment by phamhieu1412** [comment]
is same issue, please i want solution for error:((
error on oppo device

**Comment by HectorEnciso** [comment]
The faceId is only for iOS.

**Comment by devDoubleH** [comment]
Hi @MDG-MMeksasi, have you set up Face ID on your device yet?

**Comment by ferasabufares** [comment]
@MDG-MMeksasi  did you find any solution for this ???

**Comment by hodayaGruzm** [comment]
did someone has solution for this?

**Comment by ZukaGap** [comment]
Yep, I found bug and fix it. You can use my patch

https://github.com/ZukaGap/react-native-biometrics-3.0.1/tree/main

**Comment by hodayaGruzm** [comment]
@ZukaGap dont see it really working for me

**Comment by YanaSyritskaya** [comment]
@ZukaGap Hi, the patch is not working, getting [Error: Error displaying local biometric prompt: isRequiredConfirmation]. Has anybody found the solution?

**Comment by mrshahzeb7** [comment]
> Yep, I found bug and fix it. You can use my patch
> 
> https://github.com/ZukaGap/react-native-biometrics-3.0.1/tree/main

Not working for me as well

**Comment by YanaSyritskaya** [comment]
Hi all, any updates?

**Comment by usmanabid94** [comment]
Hi,
any update on this ??

**Comment by raghsahu** [comment]
Hi,
  I have used the code in ReactNativeBiometrics.java file and working fine.

   
      private PromptInfo getPromptInfo(String promptMessage, String cancelButtonText, boolean allowDeviceCredentials) {
        
        PromptInfo.Builder builder = new PromptInfo.Builder().setTitle(promptMessage).setConfirmationRequired(false);
        
        builder.setAllowedAuthenticators(getAllowedAuthenticators(allowDeviceCredentials));
        
        if (allowDeviceCredentials == false || isCurrentSDK29OrEarlier()) {
            builder.setNegativeButtonText(cancelButtonText);
        }
        return builder.build();
    }
    
     private int getAllowedAuthenticators(boolean allowDeviceCredentials) {
        
        if (allowDeviceCredentials && !isCurrentSDK29OrEarlier()) {
            return BiometricManager.Authenticators.BIOMETRIC_STRONG | BiometricManager.Authenticators.DEVICE_CREDENTIAL;
        }
        
        if (Build.VERSION.SDK_INT >= 29) {
            return BiometricManager.Authenticators.BIOMETRIC_WEAK;
        }
        
        return BiometricManager.Authenticators.BIOMETRIC_STRONG;
    }

**Comment by himanshiappzlogic** [comment]
> Hi, I have used the code in ReactNativeBiometrics.java file and working fine.
> 
> ```
>   private PromptInfo getPromptInfo(String promptMessage, String cancelButtonText, boolean allowDeviceCredentials) {
>     
>     PromptInfo.Builder builder = new PromptInfo.Builder().setTitle(promptMessage).setConfirmationRequired(false);
>     
>     builder.setAllowedAuthenticators(getAllowedAuthenticators(allowDeviceCredentials));
>     
>     if (allowDeviceCredentials == false || isCurrentSDK29OrEarlier()) {
>         builder.setNegativeButtonText(cancelButtonText);
>     }
>     return builder.build();
> }
> 
>  private int getAllowedAuthenticators(boolean allowDeviceCredentials) {
>     
>     if (allowDeviceCredentials && !isCurrentSDK29OrEarlier()) {
>         return BiometricManager.Authenticators.BIOMETRIC_STRONG | BiometricManager.Authenticators.DEVICE_CREDENTIAL;
>     }
>     
>     if (Build.VERSION.SDK_INT >= 29) {
>         return BiometricManager.Authenticators.BIOMETRIC_WEAK;
>     }
>     
>     return BiometricManager.Authenticators.BIOMETRIC_STRONG;
> }
> ```

Is this really working?

**Comment by usmanabid94** [comment]
> > Hi, I have used the code in ReactNativeBiometrics.java file and working fine.
> > ```
> >   private PromptInfo getPromptInfo(String promptMessage, String cancelButtonText, boolean allowDeviceCredentials) {
> >     
> >     PromptInfo.Builder builder = new PromptInfo.Builder().setTitle(promptMessage).setConfirmationRequired(false);
> >     
> >     builder.setAllowedAuthenticators(getAllowedAuthenticators(allowDeviceCredentials));
> >     
> >     if (allowDeviceCredentials == false || isCurrentSDK29OrEarlier()) {
> >         builder.setNegativeButtonText(cancelButtonText);
> >     }
> >     return builder.build();
> > }
> > 
> >  private int getAllowedAuthenticators(boolean allowDeviceCredentials) {
> >     
> >     if (allowDeviceCredentials && !isCurrentSDK29OrEarlier()) {
> >         return BiometricManager.Authenticators.BIOMETRIC_STRONG | BiometricManager.Authenticators.DEVICE_CREDENTIAL;
> >     }
> >     
> >     if (Build.VERSION.SDK_INT >= 29) {
> >         return BiometricManager.Authenticators.BIOMETRIC_WEAK;
> >     }
> >     
> >     return BiometricManager.Authenticators.BIOMETRIC_STRONG;
> > }
> > ```
> 
> Is this really working?

Yes I am also using this and my self a custom solution for this rather than relying on this.
